source/Xunit.Gherkin.Quick.UnitTests/GherkinScenarioOutlineTests.cs

```diff
@@ -307,12 +307,578 @@ public void ApplyExampleRow_Digests_Row_Values_Into_Scenario_With_DocString_In_S
+                            new Gherkin.Ast.TableCell(null, "quoted |person\"s^ impression"),
+                        }),
+                        new Gherkin.Ast.TableRow[]
+                        {
+                            new Gherkin.Ast.TableRow(null, new Gherkin.Ast.TableCell[]
+                            {
+                                new Gherkin.Ast.TableCell(null, "Johnny"),
+                                new Gherkin.Ast.TableCell(null, "84.444512"),
+                                new Gherkin.Ast.TableCell(null, "100 000 000"),
+                                new Gherkin.Ast.TableCell(null, "T R U E!"),
+                                new Gherkin.Ast.TableCell(null, "\"happy^| =)\""),
+                            })
+                        })
+                });
+
+            //act.
+            var scenario = sut.ApplyExampleRow("existing example", 0);
+
+            //assert.
+            Assert.NotNull(scenario);
+            Assert.Equal(sut.Name, scenario.Name);
+            Assert.Equal(sut.Steps.Count(), scenario.Steps.Count());
+            Assert.Equal(3, scenario.Steps.Count());
+
+            var sutSteps = sut.Steps.ToList();
+            var scenarioSteps = scenario.Steps.ToList();
+
+            ValidateStep(scenarioSteps[0], "Given", "Johnny compares 84.444512 with 100 000 000 via operator \"<\": \"84.444512 < 100 000 000\"", sutSteps[0]);
+            ValidateStep(scenarioSteps[1], "When", "84.444512 is being compared (not using > operator!) to 100 000 000>", sutSteps[1]);
+            ValidateStep(scenarioSteps[2], "Then", "result is T R U E!, Johnny is \"happy^| =)\" and both <84.444512> and <100 000 000> are correctly parsed", sutSteps[2]);
+
+            void ValidateStep(Gherkin.Ast.Step step, string keyword, string text, Gherkin.Ast.Step other)
+            {
+                Assert.NotSame(other, step);
+                Assert.NotSame(other.Text, step.Text);
+
+                Assert.Equal(keyword, step.Keyword);
+                Assert.Equal(text, step.Text);
+            }
+        }
+
+        [Theory]
+        [InlineData("space inbetween", "value is <space inbetween>", "value is 1")]
+        [InlineData("spaces in between", "<spaces in between>value is <spaces in between>", "1value is 1")]
+        [InlineData("special^cHaRaCtErS*BetWeeN^OR|and&plus+minus-mult*div/escape\\quote'end", "value is <special^cHaRaCtErS*BetWeeN^OR|and&plus+minus-mult*div/escape\\quote'end>", "value is 1")]
+        [InlineData("a^a", "value is <a^a>", "value is 1")]
+        [InlineData("a", "value is <a<a>", "value is <a1")]
+        [InlineData("a", "value is <a>a>", "value is 1a>")]
+        [InlineData("a|^|a", "value is <a|^|a>", "value is 1")]
+        [InlineData("a", "value is <a> <---> <a> <-> <a> < <a> > <a>", "value is 1 <---> 1 <-> 1 < 1 > 1")]
+        public void ApplyExampleRow_Digests_Row_Values_Into_Scenario_Correctly_Steps_DataTables_DocStrings(
```

Comment:
Testing how it behaves with special characters in the middle, including | and ^ (there was a bug in the previous version).